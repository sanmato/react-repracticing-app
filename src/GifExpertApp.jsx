import React, { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Kimetsu no Yaiba"]);

  const onAddCategory = (category) => {
    if (categories.includes(category)) return;
    setCategories([...categories, category]);
  };

  return (
    <div>
      <h1>GifExpertApp</h1>
      <AddCategory onNewCategory={(value) => onAddCategory(value)} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};
