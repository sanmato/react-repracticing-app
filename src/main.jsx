import { createRoot } from "react-dom/client";
import { GifExpertApp } from "./GifExpertApp";
import "./styles.css";

createRoot(document.getElementById("root")).render(<GifExpertApp />);
