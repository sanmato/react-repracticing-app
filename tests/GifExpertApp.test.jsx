import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Pruebas en GifExpertApp", () => {
  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar el título correctamente", () => {
    render(<GifExpertApp />);
    expect(screen.getByText("GifExpertApp")).toBeTruthy();
  });

  test("debe de mostrar una categoría por defecto", () => {
    render(<GifExpertApp />);
    expect(screen.getByText("Kimetsu no Yaiba")).toBeTruthy();
  });
});
