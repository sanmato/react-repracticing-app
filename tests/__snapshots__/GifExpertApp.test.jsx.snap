// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe de hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1>
      GifExpertApp
    </h1>
    <form
      aria-label="form"
    >
      <input
        placeholder="Buscar GIFs"
        type="text"
        value=""
      />
    </form>
    <h3>
      Kimetsu no Yaiba
    </h3>
    <h2>
      Cargando...
    </h2>
    <div
      class="card-grid"
    />
  </div>
</div>
`;
